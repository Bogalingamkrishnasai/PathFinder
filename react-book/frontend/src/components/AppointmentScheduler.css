/* Base Styles */
.appointment-container {
    max-width: 800px;
    margin: 2rem auto;
    padding: 2.5rem;
    background: white;
    border-radius: 16px;
    box-shadow: 0 10px 30px rgba(0, 0, 0, 0.08);
    animation: fadeIn 0.5s ease-out;
  }
  
  @keyframes fadeIn {
    from { opacity: 0; transform: translateY(20px); }
    to { opacity: 1; transform: translateY(0); }
  }
  
  @keyframes pulse {
    0% { box-shadow: 0 0 0 0 rgba(0, 123, 255, 0.4); }
    70% { box-shadow: 0 0 0 10px rgba(0, 123, 255, 0); }
    100% { box-shadow: 0 0 0 0 rgba(0, 123, 255, 0); }
  }
  
  /* Typography */
  .appointment-container h2 {
    color: #2c3e50;
    text-align: center;
    margin-bottom: 2rem;
    font-size: 2rem;
    position: relative;
  }
  
  .appointment-container h2::after {
    content: '';
    display: block;
    width: 60px;
    height: 4px;
    background: linear-gradient(90deg, #3498db, #9b59b6);
    margin: 0.5rem auto 0;
    border-radius: 2px;
  }
  
  /* Form Elements */
  .appointment-container form {
    display: flex;
    flex-direction: column;
    gap: 1.5rem;
  }
  
  .form-group {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
  }
  
  .form-group label {
    font-weight: 600;
    color: #34495e;
    font-size: 0.95rem;
  }
  
  .form-control {
    padding: 0.8rem 1rem;
    border: 2px solid #ecf0f1;
    border-radius: 8px;
    font-size: 1rem;
    transition: all 0.3s;
    background-color: #f8f9fa;
    width: 100%;
  }
  
  .form-control:focus {
    outline: none;
    border-color: #3498db;
    box-shadow: 0 0 0 3px rgba(52, 152, 219, 0.2);
  }
  
  /* Time Slots */
  .time-slots {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(100px, 1fr));
    gap: 0.8rem;
    margin-top: 0.5rem;
  }
  
  .time-slot {
    padding: 0.6rem;
    border: none;
    border-radius: 6px;
    background-color: #ecf0f1;
    color: #2c3e50;
    cursor: pointer;
    transition: all 0.2s;
    text-align: center;
    font-size: 0.9rem;
  }
  
  .time-slot:hover {
    background-color: #dfe6e9;
  }
  
  .time-slot.selected {
    background-color: #3498db;
    color: white;
    font-weight: 600;
  }
  
  .time-slot:focus {
    outline: none;
    animation: pulse 1s;
  }
  
  /* Buttons */
  .btn {
    padding: 1rem;
    color: white;
    border: none;
    border-radius: 8px;
    font-size: 1rem;
    font-weight: 600;
    cursor: pointer;
    transition: all 0.3s;
  }
  
  .btn-primary {
    background: linear-gradient(90deg, #3498db, #9b59b6);
    margin-top: 1rem;
  }
  
  .btn-primary:hover {
    transform: translateY(-2px);
    box-shadow: 0 5px 15px rgba(0, 0, 0, 0.1);
  }
  
  .btn-primary:active {
    transform: translateY(0);
  }
  
  /* Confirmation Screen */
  .confirmation-container {
    max-width: 800px;
    margin: 2rem auto;
    padding: 2.5rem;
    background: white;
    border-radius: 16px;
    box-shadow: 0 10px 30px rgba(0, 0, 0, 0.08);
    text-align: center;
    animation: fadeIn 0.5s ease-out;
  }
  
  .confirmation-details {
    background: #f8f9fa;
    padding: 1.5rem;
    border-radius: 12px;
    margin: 1.5rem 0;
    text-align: left;
    border-left: 4px solid #3498db;
  }
  
  .confirmation-details p {
    margin-bottom: 0.8rem;
    line-height: 1.6;
  }
  
  .confirmation-details strong {
    color: #2c3e50;
    display: inline-block;
    min-width: 80px;
  }
  
  /* Responsive Adjustments */
  @media (max-width: 768px) {
    .appointment-container, .confirmation-container {
      padding: 1.5rem;
      margin: 1rem;
    }
    
    .time-slots {
      grid-template-columns: repeat(auto-fill, minmax(80px, 1fr));
    }
  }
  
  @media (max-width: 480px) {
    .appointment-container h2 {
      font-size: 1.5rem;
    }
    
    .time-slots {
      grid-template-columns: repeat(2, 1fr);
    }
  }