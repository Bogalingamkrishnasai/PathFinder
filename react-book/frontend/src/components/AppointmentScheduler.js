import React, { useState } from 'react';
import './AppointmentScheduler.css';

function AppointmentScheduler() {
  const mentors = [
    { id: '1', name: 'Dr. Sarah Johnson', specialty: 'Programming Fundamentals' },
    { id: '2', name: 'Prof. Mark Williams', specialty: 'Web Development' },
    { id: '3', name: 'Dr. Lisa Chen', specialty: 'Data Structures & Algorithms' },
    { id: '4', name: 'Mr. David Kim', specialty: 'Cloud Computing' },
    { id: '5', name: 'Ms. Emily Rodriguez', specialty: 'Cybersecurity' },
    { id: '6', name: 'Dr. James Wilson', specialty: 'AI & Machine Learning' },
  ];

  const standardTimeSlots = [
    '09:00 AM', '10:00 AM', '11:00 AM',
    '12:00 PM', '01:00 PM', '02:00 PM',
    '03:00 PM', '04:00 PM',
  ];

  const [selectedMentor, setSelectedMentor] = useState('');
  const [appointmentDate, setAppointmentDate] = useState('');
  const [selectedSlot, setSelectedSlot] = useState('');
  const [appointmentNotes, setAppointmentNotes] = useState('');
  const [isSubmitted, setIsSubmitted] = useState(false);

  const handleSubmit = (e) => {
    e.preventDefault();
    if (!selectedMentor || !appointmentDate || !selectedSlot) {
      // eslint-disable-next-line
      alert('Please select a mentor, date, and time slot');
      return;
    }
    setIsSubmitted(true);
  };

  const resetForm = () => {
    setSelectedMentor('');
    setAppointmentDate('');
    setSelectedSlot('');
    setAppointmentNotes('');
    setIsSubmitted(false);
  };

  if (isSubmitted) {
    const mentor = mentors.find((m) => m.id === selectedMentor);
    return (
      <div className="confirmation-container">
        <h2>Appointment Scheduled Successfully!</h2>
        <div className="confirmation-details">
          <p>
            <strong>Mentor:</strong>
            {' '}
            {mentor?.name}
          </p>
          <p>
            <strong>Specialty:</strong>
            {' '}
            {mentor?.specialty}
          </p>
          <p>
            <strong>Date:</strong>
            {' '}
            {new Date(appointmentDate).toLocaleDateString()}
          </p>
          <p>
            <strong>Time:</strong>
            {' '}
            {selectedSlot}
          </p>
          {appointmentNotes && (
            <p>
              <strong>Notes:</strong>
              {' '}
              {appointmentNotes}
            </p>
          )}
        </div>
        <button type="button" onClick={resetForm} className="btn btn-primary">
          Schedule Another Appointment
        </button>
      </div>
    );
  }

  return (
    <div className="appointment-container">
      <h2>Schedule Appointment with Mentor</h2>

      <form onSubmit={handleSubmit}>
        <div className="form-group">
          <label htmlFor="mentor-select">
            Select Mentor:
            <select
              id="mentor-select"
              className="form-control"
              value={selectedMentor}
              onChange={(e) => setSelectedMentor(e.target.value)}
              required
            >
              <option value="">-- Select a Mentor --</option>
              {mentors.map((mentor) => (
                <option key={mentor.id} value={mentor.id}>
                  {mentor.name}
                  {' '}
                  -
                  {' '}
                  {mentor.specialty}
                </option>
              ))}
            </select>
          </label>
        </div>

        <div className="form-group">
          <label htmlFor="appointment-date">
            Appointment Date:
            <input
              type="date"
              id="appointment-date"
              className="form-control"
              value={appointmentDate}
              onChange={(e) => setAppointmentDate(e.target.value)}
              min={new Date().toISOString().split('T')[0]}
              required
            />
          </label>
        </div>

        {appointmentDate && (
          <div className="form-group">
            <label htmlFor="time-slots">
              Available Time Slots:
              <div className="time-slots">
                {standardTimeSlots.map((slot) => (
                  <button
                    type="button"
                    key={slot}
                    className={`time-slot ${selectedSlot === slot ? 'selected' : ''}`}
                    onClick={() => setSelectedSlot(slot)}
                    onKeyDown={(e) => {
                      if (e.key === 'Enter' || e.key === ' ') {
                        setSelectedSlot(slot);
                      }
                    }}
                  >
                    {slot}
                  </button>
                ))}
              </div>
            </label>
          </div>
        )}

        <div className="form-group">
          <label htmlFor="appointment-notes">
            Notes (Optional):
            <textarea
              id="appointment-notes"
              className="form-control"
              rows="3"
              value={appointmentNotes}
              onChange={(e) => setAppointmentNotes(e.target.value)}
              placeholder="Any specific topics you'd like to discuss..."
            />
          </label>
        </div>

        <button type="submit" className="btn btn-primary">
          Schedule Appointment
        </button>
      </form>
    </div>
  );
}

export default AppointmentScheduler;
