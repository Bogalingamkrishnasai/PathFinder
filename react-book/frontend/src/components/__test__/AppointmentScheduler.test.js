import { render, screen, fireEvent } from '@testing-library/react';
import AppointmentScheduler from '../AppointmentScheduler';

// Mock CSS imports
jest.mock('../AppointmentScheduler.css', () => ({}));

describe('AppointmentScheduler Component', () => {
  const mockMentors = [
    { id: '1', name: 'Dr. Sarah Johnson', specialty: 'Programming Fundamentals' },
    { id: '2', name: 'Prof. Mark Williams', specialty: 'Web Development' }
  ];

  const renderComponent = () => {
    return render(<AppointmentScheduler mentors={mockMentors} />);
  };

  test('renders scheduler form', () => {
    const { asFragment } = renderComponent();
    expect(screen.getByText('Schedule Appointment with Mentor')).toBeInTheDocument();
    expect(screen.getByLabelText('Select Mentor:')).toBeInTheDocument();
    expect(screen.getByLabelText('Appointment Date:')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('shows validation errors when submitting empty form', () => {
    const { asFragment } = renderComponent();
    const submitButton = screen.getByText('Schedule Appointment');
    fireEvent.click(submitButton);
    
    // Check for browser-native validation messages
    const mentorSelect = screen.getByLabelText('Select Mentor:');
    const dateInput = screen.getByLabelText('Appointment Date:');
    
    expect(mentorSelect).toBeInvalid();
    expect(dateInput).toBeInvalid();
    expect(asFragment()).toMatchSnapshot();
  });

  test('shows time slots after date is selected', () => {
    const { asFragment } = renderComponent();
    const dateInput = screen.getByLabelText('Appointment Date:');
    fireEvent.change(dateInput, { target: { value: '2024-12-31' } });
    
    expect(screen.getByText('Available Time Slots:')).toBeInTheDocument();
    expect(screen.getAllByRole('button', { name: /AM|PM/ }).length).toBeGreaterThan(0);
    expect(asFragment()).toMatchSnapshot();
  });

  test('shows confirmation after successful submission', () => {
    const { asFragment } = renderComponent();
    
    // Fill out form
    fireEvent.change(screen.getByLabelText('Select Mentor:'), { 
      target: { value: '1' } 
    });
    fireEvent.change(screen.getByLabelText('Appointment Date:'), { 
      target: { value: '2024-12-31' } 
    });
    fireEvent.click(screen.getAllByRole('button', { name: /AM|PM/ })[0]);
    
    // Submit
    fireEvent.click(screen.getByText('Schedule Appointment'));
    
    expect(screen.getByText('Appointment Scheduled Successfully!')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  // Additional snapshots
  test('matches empty form snapshot', () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  test('matches time slots selection snapshot', () => {
    const { asFragment } = renderComponent();
    fireEvent.change(screen.getByLabelText('Appointment Date:'), { 
      target: { value: '2024-12-31' } 
    });
    expect(asFragment()).toMatchSnapshot();
  });
});