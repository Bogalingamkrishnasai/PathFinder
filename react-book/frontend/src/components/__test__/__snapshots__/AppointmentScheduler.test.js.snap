// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentScheduler Component matches empty form snapshot 1`] = `
<DocumentFragment>
  <div
    class="appointment-container"
  >
    <h2>
      Schedule Appointment with Mentor
    </h2>
    <form>
      <div
        class="form-group"
      >
        <label
          for="mentor-select"
        >
          Select Mentor:
          <select
            class="form-control"
            id="mentor-select"
            required=""
          >
            <option
              value=""
            >
              -- Select a Mentor --
            </option>
            <option
              value="1"
            >
              Dr. Sarah Johnson - Programming Fundamentals
            </option>
            <option
              value="2"
            >
              Prof. Mark Williams - Web Development
            </option>
            <option
              value="3"
            >
              Dr. Lisa Chen - Data Structures & Algorithms
            </option>
            <option
              value="4"
            >
              Mr. David Kim - Cloud Computing
            </option>
            <option
              value="5"
            >
              Ms. Emily Rodriguez - Cybersecurity
            </option>
            <option
              value="6"
            >
              Dr. James Wilson - AI & Machine Learning
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-date"
        >
          Appointment Date:
          <input
            class="form-control"
            id="appointment-date"
            min="2025-04-23"
            required=""
            type="date"
            value=""
          />
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-notes"
        >
          Notes (Optional):
          <textarea
            class="form-control"
            id="appointment-notes"
            placeholder="Any specific topics you'd like to discuss..."
            rows="3"
          />
        </label>
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Schedule Appointment
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`AppointmentScheduler Component matches time slots selection snapshot 1`] = `
<DocumentFragment>
  <div
    class="appointment-container"
  >
    <h2>
      Schedule Appointment with Mentor
    </h2>
    <form>
      <div
        class="form-group"
      >
        <label
          for="mentor-select"
        >
          Select Mentor:
          <select
            class="form-control"
            id="mentor-select"
            required=""
          >
            <option
              value=""
            >
              -- Select a Mentor --
            </option>
            <option
              value="1"
            >
              Dr. Sarah Johnson - Programming Fundamentals
            </option>
            <option
              value="2"
            >
              Prof. Mark Williams - Web Development
            </option>
            <option
              value="3"
            >
              Dr. Lisa Chen - Data Structures & Algorithms
            </option>
            <option
              value="4"
            >
              Mr. David Kim - Cloud Computing
            </option>
            <option
              value="5"
            >
              Ms. Emily Rodriguez - Cybersecurity
            </option>
            <option
              value="6"
            >
              Dr. James Wilson - AI & Machine Learning
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-date"
        >
          Appointment Date:
          <input
            class="form-control"
            id="appointment-date"
            min="2025-04-23"
            required=""
            type="date"
            value="2024-12-31"
          />
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="time-slots"
        >
          Available Time Slots:
          <div
            class="time-slots"
          >
            <button
              class="time-slot "
              type="button"
            >
              09:00 AM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              10:00 AM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              11:00 AM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              12:00 PM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              01:00 PM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              02:00 PM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              03:00 PM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              04:00 PM
            </button>
          </div>
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-notes"
        >
          Notes (Optional):
          <textarea
            class="form-control"
            id="appointment-notes"
            placeholder="Any specific topics you'd like to discuss..."
            rows="3"
          />
        </label>
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Schedule Appointment
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`AppointmentScheduler Component renders scheduler form 1`] = `
<DocumentFragment>
  <div
    class="appointment-container"
  >
    <h2>
      Schedule Appointment with Mentor
    </h2>
    <form>
      <div
        class="form-group"
      >
        <label
          for="mentor-select"
        >
          Select Mentor:
          <select
            class="form-control"
            id="mentor-select"
            required=""
          >
            <option
              value=""
            >
              -- Select a Mentor --
            </option>
            <option
              value="1"
            >
              Dr. Sarah Johnson - Programming Fundamentals
            </option>
            <option
              value="2"
            >
              Prof. Mark Williams - Web Development
            </option>
            <option
              value="3"
            >
              Dr. Lisa Chen - Data Structures & Algorithms
            </option>
            <option
              value="4"
            >
              Mr. David Kim - Cloud Computing
            </option>
            <option
              value="5"
            >
              Ms. Emily Rodriguez - Cybersecurity
            </option>
            <option
              value="6"
            >
              Dr. James Wilson - AI & Machine Learning
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-date"
        >
          Appointment Date:
          <input
            class="form-control"
            id="appointment-date"
            min="2025-04-23"
            required=""
            type="date"
            value=""
          />
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-notes"
        >
          Notes (Optional):
          <textarea
            class="form-control"
            id="appointment-notes"
            placeholder="Any specific topics you'd like to discuss..."
            rows="3"
          />
        </label>
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Schedule Appointment
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`AppointmentScheduler Component shows confirmation after successful submission 1`] = `
<DocumentFragment>
  <div
    class="confirmation-container"
  >
    <h2>
      Appointment Scheduled Successfully!
    </h2>
    <div
      class="confirmation-details"
    >
      <p>
        <strong>
          Mentor:
        </strong>
         Dr. Sarah Johnson
      </p>
      <p>
        <strong>
          Specialty:
        </strong>
         Programming Fundamentals
      </p>
      <p>
        <strong>
          Date:
        </strong>
         12/30/2024
      </p>
      <p>
        <strong>
          Time:
        </strong>
         09:00 AM
      </p>
    </div>
    <button
      class="btn btn-primary"
      type="button"
    >
      Schedule Another Appointment
    </button>
  </div>
</DocumentFragment>
`;

exports[`AppointmentScheduler Component shows time slots after date is selected 1`] = `
<DocumentFragment>
  <div
    class="appointment-container"
  >
    <h2>
      Schedule Appointment with Mentor
    </h2>
    <form>
      <div
        class="form-group"
      >
        <label
          for="mentor-select"
        >
          Select Mentor:
          <select
            class="form-control"
            id="mentor-select"
            required=""
          >
            <option
              value=""
            >
              -- Select a Mentor --
            </option>
            <option
              value="1"
            >
              Dr. Sarah Johnson - Programming Fundamentals
            </option>
            <option
              value="2"
            >
              Prof. Mark Williams - Web Development
            </option>
            <option
              value="3"
            >
              Dr. Lisa Chen - Data Structures & Algorithms
            </option>
            <option
              value="4"
            >
              Mr. David Kim - Cloud Computing
            </option>
            <option
              value="5"
            >
              Ms. Emily Rodriguez - Cybersecurity
            </option>
            <option
              value="6"
            >
              Dr. James Wilson - AI & Machine Learning
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-date"
        >
          Appointment Date:
          <input
            class="form-control"
            id="appointment-date"
            min="2025-04-23"
            required=""
            type="date"
            value="2024-12-31"
          />
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="time-slots"
        >
          Available Time Slots:
          <div
            class="time-slots"
          >
            <button
              class="time-slot "
              type="button"
            >
              09:00 AM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              10:00 AM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              11:00 AM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              12:00 PM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              01:00 PM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              02:00 PM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              03:00 PM
            </button>
            <button
              class="time-slot "
              type="button"
            >
              04:00 PM
            </button>
          </div>
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-notes"
        >
          Notes (Optional):
          <textarea
            class="form-control"
            id="appointment-notes"
            placeholder="Any specific topics you'd like to discuss..."
            rows="3"
          />
        </label>
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Schedule Appointment
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`AppointmentScheduler Component shows validation errors when submitting empty form 1`] = `
<DocumentFragment>
  <div
    class="appointment-container"
  >
    <h2>
      Schedule Appointment with Mentor
    </h2>
    <form>
      <div
        class="form-group"
      >
        <label
          for="mentor-select"
        >
          Select Mentor:
          <select
            class="form-control"
            id="mentor-select"
            required=""
          >
            <option
              value=""
            >
              -- Select a Mentor --
            </option>
            <option
              value="1"
            >
              Dr. Sarah Johnson - Programming Fundamentals
            </option>
            <option
              value="2"
            >
              Prof. Mark Williams - Web Development
            </option>
            <option
              value="3"
            >
              Dr. Lisa Chen - Data Structures & Algorithms
            </option>
            <option
              value="4"
            >
              Mr. David Kim - Cloud Computing
            </option>
            <option
              value="5"
            >
              Ms. Emily Rodriguez - Cybersecurity
            </option>
            <option
              value="6"
            >
              Dr. James Wilson - AI & Machine Learning
            </option>
          </select>
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-date"
        >
          Appointment Date:
          <input
            class="form-control"
            id="appointment-date"
            min="2025-04-23"
            required=""
            type="date"
            value=""
          />
        </label>
      </div>
      <div
        class="form-group"
      >
        <label
          for="appointment-notes"
        >
          Notes (Optional):
          <textarea
            class="form-control"
            id="appointment-notes"
            placeholder="Any specific topics you'd like to discuss..."
            rows="3"
          />
        </label>
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Schedule Appointment
      </button>
    </form>
  </div>
</DocumentFragment>
`;
